common/src/main/java/com/wynntils/screens/OverlayManagementScreen.java

```diff
@@ -0,0 +1,651 @@
+                    minY = overlay.getRenderY();
+                    maxY = overlay.getRenderY() + overlay.getHeight();
+                }
+                case Bottom -> {
+                    minX = overlay.getRenderX();
+                    maxX = overlay.getRenderX() + overlay.getWidth();
+                    minY = overlay.getRenderY() + overlay.getHeight() - MAX_CLICK_DISTANCE / 2f;
+                    maxY = overlay.getRenderY() + overlay.getHeight() + MAX_CLICK_DISTANCE / 2f;
+                }
+                default -> {
+                    // should not happen
+                    continue;
+                }
+            }
+
+            if (mouseX >= minX && mouseX <= maxX && mouseY >= minY && mouseY <= maxY) {
+                selectedEdge = value;
+                selectionMode = SelectionMode.Edge;
+
+                return false;
+            }
+        }
+
+        if (isMouseHoveringOverlay(selectedOverlay, mouseX, mouseY)) {
+            selectionMode = SelectionMode.Area;
+
+            return false;
+        }
+
+        return super.mouseClicked(mouseX, mouseY, button);
+    }
+
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        if (testMode) return false;
+
+        resetSelection();
+        return super.mouseReleased(mouseX, mouseY, button);
+    }
+
+    @Override
+    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+        if (testMode) return false;
+
+        if (keyCode == GLFW.GLFW_KEY_UP || keyCode == GLFW.GLFW_KEY_DOWN) {
+            int index = selectedOverlay.getRenderVerticalAlignment().ordinal();
+
+            if (keyCode == GLFW.GLFW_KEY_DOWN) {
+                index += 1;
+            } else {
+                index -= 1;
+            }
+
+            VerticalAlignment[] values = VerticalAlignment.values();
+            index = (values.length + index) % values.length;
+            selectedOverlay.setVerticalAlignmentOverride(values[index]);
+            saveConfigForOverlay();
+        } else if (keyCode == GLFW.GLFW_KEY_RIGHT || keyCode == GLFW.GLFW_KEY_LEFT) {
+            int index = selectedOverlay.getRenderHorizontalAlignment().ordinal();
+
+            if (keyCode == GLFW.GLFW_KEY_RIGHT) {
+                index += 1;
+            } else {
+                index -= 1;
+            }
+
+            HorizontalAlignment[] values = HorizontalAlignment.values();
+            index = (values.length + index) % values.length;
+            selectedOverlay.setHorizontalAlignmentOverride(values[index]);
+            saveConfigForOverlay();
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean mouseDragged(double mouseX, double mouseY, int button, double dragX, double dragY) {
+        if (testMode) return false;
+
+        // Order:
+        //  - Corners
+        //  - Edges
+        //  - OverlayArea
+
+        handleOverlayCornerDrag(button, dragX, dragY);
```

Comment:
Maybe a switch on `selectionMode` instead of just calling all three blindly? More for a readability thing, I realize the code works as well when written like this.